src/test/java/robot/subsystems/drivetrain/SwerveDriveTest.java

```diff
@@ -0,0 +1,102 @@
+package robot.subsystems.drivetrain;
+
+import org.junit.*;
+
+import static robot.Constants.SwerveDrive.*;
+
+public class SwerveDriveTest {
+
+    private SwerveDrive swerveDrive;
+    private SwerveDrive swerveField;
+
+    private double forward, strafe, rotation, gyro;
+    private double deviation = 0.01;
+    private double[] expectedHeading, expectedHeadingField, expectedVel,
+            expectedVelField, robotHeading, robotHeadingField;
+
+    @Before
+    public void setup() {
+        swerveDrive = new SwerveDrive(false);
+        swerveField = new SwerveDrive(true);
+    }
+
+    @Test
+    public void turnInPlace() {
+        forward = 0;
+        strafe = 0;
+        rotation = -1;
+
+        expectedHeading = new double[]{0, 0, rotation * ROTATION_MULTIPLIER};
+        expectedHeadingField = new double[]{0, 0, rotation * ROTATION_MULTIPLIER};
+
+        expectedVel = new double[]{-Math.PI / 2, Math.PI / 2, -Math.PI / 2, -Math.PI / 2, Math.PI / 2, -Math.PI / 2, Math.PI / 2, Math.PI / 2};
+
+        getRobotHeading();
+        getRobotHeadingField();
+        calculateWheelVelocities();
+    }
+
+    @Test
+    public void driveForward() {
+        forward = 1;
+        strafe = 0;
+        rotation = 0;
+        gyro =  3 * Math.PI / 2;
+
+        expectedHeading = new double[]{forward * SPEED_MULTIPLIER, 0, 0};

```

Comment:
I suggest that you do enter the `rotation` and `strafe` into the robot heading so it would be clearer